### PR TITLE
feat: return Table from `create_table()` and `create_view()`

### PR DESCRIPTION
**Pull request by NickCrews**
Closes https://github.com/ibis-project/ibis/issues/5693

I didn't add any tests sepcific to this,
because I modified some existing tests so that
they relied on this new behavior, so this new
behavior is getting tested as a side benefit.
The tests are also clearer now because of this.
I did NOT adjust some tests that I thought would
more clear to have two separate steps of
creating the table/view and accessing it later.

### CODE REVIEW

**Review by jcrist — APPROVED** [review]
Thanks @NickCrews! Overall this looks good to me, but I'd like a second set of :eyes: on it before merging (cc @cpcloud).

File: ibis/backends/base/sql/alchemy/__init__.py [hunk 1]
```diff
@@ -174,23 +174,32 @@ def create_table(
...
         """Create a table.
 
         Parameters
         ----------
         name
-            Table name to create
+            Name of the new table.
         expr
-            DataFrame or table expression to use as the data source
+            An Ibis table expression or pandas table that will be used to
```
Comment (NickCrews) [inline ibis/backends/base/sql/alchemy/__init__.py:185]: copied from docstring of super()

File: ibis/backends/bigquery/__init__.py [hunk 2]
```diff
@@ -460,6 +460,7 @@ def create_table(
             sql_select = self.compile(table)
             table_ref = f"`{project_id}`.`{dataset}`.`{name}`"
             self.raw_sql(f'CREATE TABLE {table_ref} AS ({sql_select})')
+        return self.table(name, database=database)
```
Comment (NickCrews) [inline ibis/backends/bigquery/__init__.py:463]: codecov says this isn't covered, but this is unrelated to this change. Open an issue to add a test for create_table() on bigquery?

File: ibis/backends/dask/__init__.py [hunk 3]
```diff
@@ -128,15 +128,6 @@ def _convert_schema(schema: sch.Schema):
     def _convert_object(cls, obj: dd.DataFrame) -> dd.DataFrame:
         return obj
 
-    def create_table(
-        self,
-        table_name: str,
-        obj: dd.DataFrame | None = None,
```
Comment (NickCrews) [inline ibis/backends/dask/__init__.py:134]: Actually, I guess overriding this method made it so we could have different type annotations from super(). Should this not get removed so we keep the custom annotations?

File: ibis/backends/bigquery/__init__.py (hunk 2 above)
Comment (cpcloud) [reply ibis/backends/bigquery/__init__.py:463]: It's reporting no coverage because we don't run the bigquery tests until after merge, because we don't want PRs to have access to creds.

Comment (cpcloud) [reply ibis/backends/bigquery/__init__.py:463]: No need for an issue right now.
